### PR TITLE
feat: sync CRD and ingress resource to apisix mechanism.

### PR DESCRIPTION
**Pull request by AlinsRan**
### Type of change:

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### Pre-submission checklist:
**Quote:**
* #234 
* #1022 

Resolve: #1099 

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by tao12345666333** [comment]
could you merge latest code? thanks!

**Comment by AlinsRan** [comment]
> could you merge latest code? thanks!

Done.

File: pkg/ingress/ingress.go [hunk 1]
```diff
@@ -403,3 +403,25 @@ func (c *ingressController) isIngressEffective(ing kube.Ingress) bool {
 	}
 	return false
 }
+
+func (c *ingressController) ResourceSync() {
+	objs := c.controller.ingressInformer.GetIndexer().List()
+	for _, obj := range objs {
+		key, err := cache.MetaNamespaceKeyFunc(obj)
+		if err != nil {
+			log.Errorw("found ApisixConsumer resource with bad meta namespace key", zap.String("error", err.Error()))
```
Comment (tao12345666333) [inline pkg/ingress/ingress.go:412]: It's not ApisixConsumer

**Review by tao12345666333 — APPROVED** [review]
LGTM

thanks

**Review by lingsamuel — APPROVED** [review]

**Pull request merged**